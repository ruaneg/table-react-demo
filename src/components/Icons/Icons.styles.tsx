import styled from 'styled-components';

export const IconContainer = styled.span`
  padding: 0 8px;
  font-size: 0.875rem;
`;
