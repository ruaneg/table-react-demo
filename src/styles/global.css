body {
  font-family: "Arial", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans",
    "Droid Sans", "Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;

  color: #000000d6;
}
