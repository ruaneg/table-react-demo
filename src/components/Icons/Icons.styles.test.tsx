import { render } from '@testing-library/react';
import 'jest-styled-components';
import { describe, test, expect } from 'vitest';
import { IconContainer } from './Icons.styles';

describe('IconContainer', () => {
  test('renders correctly with default styles', () => {
    const { container } = render(<IconContainer />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('applies correct styles', () => {
    const { container } = render(<IconContainer />);
    const element = container.firstChild;

    expect(element).toHaveStyleRule('padding', '0 8px');
    expect(element).toHaveStyleRule('font-size', '0.875rem');
  });
});
