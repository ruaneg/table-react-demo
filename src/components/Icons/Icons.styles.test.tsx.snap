// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconContainer > renders correctly with default styles 1`] = `
.c0 {
  padding: 0 8px;
  font-size: 0.875rem;
}

<span
  class="c0"
/>
`;
